### PR TITLE
School Library: Unit tests

### PR DESCRIPTION
**Pull request by Timbar09**
In this PR, we ( @Timbar09  and @fitz95 ) added unit tests for methods in the classes.

Changes
- Added unit tests for methods in the `Book`, `Person`, `Student`, `Teacher`,  `Classroom`, and `Rental` classes.
- Added unit tests for methods of all classes created while implementing the Decorator design pattern

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    end
+
+    it('should have an empty array of rentals') do
+      expect(test_book.rentals).to eq []
+    end
+
+    it('should have a method to_json') do
+      expect(test_book.to_json).to be_a Hash
+    end
+  end
```
Comment (TadesseDev) [inline spec/book_spec.rb]: - [ ] You did well testing your initializer for a book class, But as part of the main requirements for this milestone, we need to have an 'it' block for each class method (at least). Please add a test case for your `add_rental` (`def add_rental(person, date)`) method.

File: spec/classroom_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    classroom = Classroom.new('Math')
+
+    it 'should have a label' do
+      expect(classroom.label).to eql('Math')
+    end
+
+    it 'should have no students' do
+      expect(classroom.students).to eql([])
+    end
+  end
```
Comment (TadesseDev) [inline spec/classroom_spec.rb]: - [x] You did well testing your initializer for a Classroom class, But as part of the main requirements for this milestone, we need to have an 'it' block for each class method (at least). Please add a test case for your `add_student` (`def add_student(student))`) method.

File: spec/person_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    it('should have an empty array of rentals') do
+      expect(test_person.rentals).to eq []
+    end
+
+    it('should have a method can_use_services') do
+      expect(test_person.can_use_services?).to be true
+    end
+  end
+end
```
Comment (TadesseDev) [inline spec/person_spec.rb]: - [x] You did well testing your initializer for a Person class, But as part of the main requirements for this milestone, we need to have an 'it' block for each class method (at least). Please add a test case for your `add_rental` method inside the person class.

File: spec/student_spec.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
+require_relative 'all_methods'
+
+describe Student do
+  context 'When testing the Student class, it' do
+    physics = Classroom.new('Physics')
+    parent_permission = true
+    tokyo = Student.new(nil, physics, 16, 'Tokyo', parent_permission)
+
+    it 'should create a new student' do
+      expect(tokyo).to be_a(Student)
```
Comment (TadesseDev) [inline spec/student_spec.rb]: - [x] You did well testing some of your methods inside the Student class, But as part of the main requirements for this milestone, we need to have an 'it' block for each class method (at least). Please add a test case for your `to_json, play_hooky, and add_classroom` methods inside the Student class.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x]  Correct GitFlow  ✅
 - [x]  Good commit history  ✅
 - [x]  Good PR title/description  ✅
 - [x]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Timbar09** [comment]
Thank you, @TadesseDev for pointing out the issues. We will address them.

File: spec/book_spec.rb (hunk 1 above)
Comment (Whoistolu) [reply spec/book_spec.rb]: - Well resolved 👍🏽

File: spec/classroom_spec.rb (hunk 2 above)
Comment (Whoistolu) [reply spec/classroom_spec.rb]: - Well resolved 👍🏽

File: spec/person_spec.rb (hunk 3 above)
Comment (Whoistolu) [reply spec/person_spec.rb]: - Well resolved 👍🏽

File: spec/student_spec.rb (hunk 4 above)
Comment (Whoistolu) [reply spec/student_spec.rb]: - Well resolved 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi Team,
@Timbar09 and @fitz95 

Great job testing the rest of the methods been pointed out by the previous reviewer 👍🏽 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Timbar09** [comment]
Thank you, @Whoistolu, for approving our project and giving us clear and productive feedback.

**Pull request merged**